velox/common/base/Cleanup.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include "velox/common/base/BitUtil.h"
+
+namespace facebook {
+namespace velox {
+
+/// A scoped helper object invokes user provided cleanup callback on
```

Comment:
Any particular reason not to use folly::ScopeGuard? 

https://github.com/facebook/folly/blob/main/folly/ScopeGuard.h

Reply:
You mean folly::makeGuard? Thanks!